### PR TITLE
docs[DST-72] Adding live demo

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

File: docs/demo/components/Button/button-basic.demo.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import LiveDemoEditor from '@/DemoLiveEditor';
+import { Button } from '@marigold/components';
+
+export const BasicButton = () => {
+  return (
+    <LiveDemoEditor
+      scope={{ Button }}
+      code={`
+            <button className='text-white rounded-lg'>demo button</button>
```
Comment (sebald) [inline docs/demo/components/Button/button-basic.demo.tsx:9]: Can this work with TypeScript code too? Because this is what we want to display :)

Comment (OsamaAbdellateef) [reply docs/demo/components/Button/button-basic.demo.tsx:9]: it works fine for reading typescript and i tried  an example for this, but you can't see what types are on hover :"
is that fine ?!

Comment (OsamaAbdellateef) [reply docs/demo/components/Button/button-basic.demo.tsx:9]: you can run the app and go to `http://localhost:3000/counter`
And you can view the file for this on `docs/demo/components/Counter`

**Comment by sebald** [comment]
@OsamaAbdellateef this looks great, do you want to merge this or work in this branch for a refunded version?

PS: we need to specify what the code demi should be able to do 😀

**Comment by sebald** [comment]
@OsamaAbdellateef this looks great, do you want to merge this or work in this branch for a refunded version?

PS: we need to specify what the code demi should be able to do 😀

**Comment by OsamaAbdellateef** [comment]
> @OsamaAbdellateef this looks great, do you want to merge this or work in this branch for a refunded version?
> 
> PS: we need to specify what the code demi should be able to do grinning

Yes i think we can merge 
and i guess demo will be used  to change the logic , variants or size or even logic stuff

**Comment by OsamaAbdellateef** [comment]
> > @OsamaAbdellateef this looks great, do you want to merge this or work in this branch for a refunded version?
> > PS: we need to specify what the code demi should be able to do grinning
> 
> Yes i think we can merge and i guess demo will be used to change the logic , variants or size or even logic stuff

I'll resolve the conflicts first

**Comment by sarahgm** [comment]
@OsamaAbdellateef  Have I forget something or why does is has no style on my screen? Counter has styles

Bildschirmfoto 2023-07-06 um 09 14 22

**Comment by sarahgm** [comment]
also I get a warning in browser console: `Warning: CodeEditor: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead`

**Comment by sebald** [comment]
@OsamaAbdellateef I think we can close this?

**Comment by OsamaAbdellateef** [comment]
> @OsamaAbdellateef I think we can close this?

@sebald 
Yes Sure i will close it

**Pull request closed without merging**